app/controllers/grade_entry_forms_controller.rb

```diff
@@ -270,10 +267,54 @@ def get_mark_columns
                  + ' ' + grade_entry_form.out_of_total.to_s,
       }
     end
+    if current_user.admin? || current_user.ta?
+      c <<
+      {
+        id: 'marking_state',
+        content: t('grade_entry_forms.grades.marking_state')
+      }
+    end
 
     render json: c
   end
 
+  def populate_grades_spreadsheet_table
```

Comment:
Assignment Branch Condition size for populate_grades_spreadsheet_table is too high. [39.1/15]